### DISCUSSION TITLE
macro to support EditorConfig

### DISCUSSION THREAD

**Discussion by tksoh** [eteran/nedit-ng#301] [Show and tell]
For those who might be interested, you can checkout the macro here:

https://github.com/tksoh/nedit-macros/blob/master/editorconfig/editorconfig.nm

It's still very much work in progress, but it seems to work for nedit-ng repo so far. Right now it merely works on the tabs settings (indent_style & indent_size), since other properties are not immediately applicable to nedit. Glob patterns support is not complete, but commonly used patterns should work without issue.

It works on both nedit-ng and nedit 5.

I am aware of @eteran's plan to support EditorConfig natively in nedit-ng, but until then, or if you are still using nedit 5 somehow, this macro should work as a temp solution.

**Comment by eteran** [comment]
Fantastic! Honestly, even my native solution will probably look to this for some inspiration :-)

**Reply by tksoh** [reply]
I've been wondering if we should add some nedit-specific properties. Comment?

**Reply by eteran** [reply]
Unfortunately, I'm not sure if it's a good idea to have non-standard properties since the whole benefit of editorconfig is that it is a standard project configuration file.

**Reply by tksoh** [reply]
Having a standard is a good point. But it would mean we have to change the internal of nedit to fit the standards, which may or may not work out well. Also, getting any new property into the standard is going to be a lengthy process, assuming it's even adopted as candidate, since they are a host of editors out there, each with their own quirks and philosophy. 

EditorConfig already has section on properties that "Supported By A Limited Number of Editors", currently the 'max_line_length' property. Given that, in my view, the .editorconfig should allow the extension to support specific editors -- any editors that don't understand the non-standard properties should be able to just safely ignore them. Admittedly, that could undermine their plan (I suspect ;-) to encourage the editors to adapt to the standard properties ;-)
